app/src/main/java/com/example/next_show/fragments/FeedFragment.java

```diff
@@ -52,21 +51,34 @@
     private static final String TAG = "FeedFragment";
     private static final String SHOW_DETAIL_URL = "https://api.themoviedb.org/3/tv/";
     private static final String ADD_API_KEY = "?api_key=";
-    public static final int NOT_FOUND = -1;
+    private static final int NOT_FOUND = -1;
+
+    // lists match with dialog list of checks -> view related
+    private static final String[] GENRE_FILTERS = {ShowFilter.ACTION, ShowFilter.COMEDY, ShowFilter.DRAMA};
+    private boolean[] selectedGenres = {false, false, false};
+    private static final String[] NETWORK_FILTERS = {ShowFilter.CABLE, ShowFilter.STREAMING};
+    private boolean[] selectedNetworks = {false, false};
+    private static final String[] YEAR_FILTERS = {ShowFilter.PRIOR, ShowFilter.PAST_FIVE, ShowFilter.THIS_YEAR};
+    private boolean[] selectedYears = {false, false, false};
```

Comment:
I think it might be better, instead of keeping an array of booleans, you just have a set of selected XXXX and then add and remove from the sets.

Otherwise, in order to know what genre is selected, you also have to base it on the ordering in the other list.

Reply:
Unfortunately, I am using these arrays in the AlertDialog.Builder and the inputs to build the checkboxes are boolean[] and String[] for the labels...not sure how to keep track of a set and convert it to boolean[] when needed although I see why keeping a set is a better approach!